gateway/src/apicast/policy/upstream/upstream.lua

```diff
@@ -27,7 +27,7 @@ end
 
 local function change_upstream(url)
   ngx.ctx.upstream = resty_resolver:instance():get_servers(
-    url.host, { port = url.port })
+    url.host, { port = url.port or resty_url.default_port(url.scheme) })
```

Comment:
I think this can't be here. When a port is passed here, it overwrites the one set by the resolver.
That's why the `sanity.t` integration test fails. That test uses the `upstream` policy:
```
      "policy_chain": [
        { "name": "apicast.policy.upstream",
          "configuration": { "rules": [ { "regex": "/", "url": "http://echo" } ] } }
      ]
```

In that case, when `resty_resolver` resolves `echo`, it includes a port (randomly generated by the test framework in this case), and your solution overwrites it with `80`.

I think the port should be set only when it's not set by the resolver.

Reply:
@davidor how about https://github.com/3scale/apicast/pull/662?